src/components/EventDetails/EventDetails.js

```diff
@@ -1382,51 +1375,36 @@ export class EventDetail extends React.Component<Props> {
     const title = actionTitle || actionLabel || fullItemValue;
     const label = name || itemLabel;
     const subtitle = (actionSubtitle || fullItemValue) ? actionSubtitle || subtext : null;
-    const titleColor = this.getColor(valueColor);
-    const eventTime = date && formatDate(new Date(date * 1000), 'MMMM D, YYYY HH:mm');
+    const titleColor = this.getColor(valueColor) || undefined;
+
+    const commonProps = {
+      date: date !== undefined ? new Date(date * 1000) : undefined,
+      title: label,
+      subtitle: sublabel,
+      image: this.renderImage(itemData),
+      buttons,
+    };
 
-    return (
-      <Wrapper forceInset={{ top: 'never', bottom: 'always' }}>
-        <EventTimeHolder>
-          <BaseText tiny secondary>{eventTime}</BaseText>
-        </EventTimeHolder>
-        <Spacing h={10} />
-        <AvatarWrapper disabled>
-          <BaseText medium>{label}</BaseText>
-          {sublabel && <BaseText regular secondary>{sublabel}</BaseText>}
-          <Spacing h={20} />
-          {this.renderImage(itemData)}
-        </AvatarWrapper>
-        <Spacing h={20} />
-        {settleEventData ? this.renderSettle(settleEventData, eventData) : (
-          <React.Fragment>
-            <ActionWrapper>
-              {!!title && <MediumText large color={titleColor}>{title}</MediumText>}
-              {customActionTitle}
-              {!!actionIcon && <ActionIcon name={actionIcon} iconColor={statusIconColor} />}
-            </ActionWrapper>
-            {subtitle ? (
-              <React.Fragment>
-                <Spacing h={4} />
-                <BaseText regular secondary>{subtitle}</BaseText>
-                <Spacing h={16} />
-              </React.Fragment>
-            ) : (
-              <Spacing h={32} />
-            )}
-            {!!errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
-            {this.renderFee(event.hash, fee, isReceived)}
-          </React.Fragment>
+    return settleEventData ? (
+      <DetailModal {...commonProps}>
+        {this.renderSettle(settleEventData, eventData)}
+      </DetailModal>
+    ) : (
+      <DetailModal
+        {...commonProps}
+        fee={this.getFee(event.hash, fee, isReceived)}
+      >
+        {!!title && (
+          <DetailRow color={titleColor}>
+            {title}
+            {!!actionIcon && <ActionIcon name={actionIcon} iconColor={statusIconColor} />}
+          </DetailRow>
         )}
-        <ButtonsContainer>
-          {buttons.map(buttonProps => (
-            <React.Fragment key={buttonProps.title} >
-              <Button regularText {...buttonProps} />
-              <Spacing h={4} />
-            </React.Fragment>
-          ))}
-        </ButtonsContainer>
-      </Wrapper>
+        {customActionTitle}
+
+        {!!subtitle && <DetailParagraph>{subtitle}</DetailParagraph>}
+        {!!errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
+      </DetailModal>
```

Comment:
Just an idea:
since it's return, maybe this would be more readable?

``` 
    if (settleEventData) {
      return (
        <DetailModal {...commonProps}>
          {this.renderSettle(settleEventData, eventData)}
        </DetailModal>
      );
    }

    return (
      <DetailModal
        {...commonProps}
        fee={this.getFee(event.hash, fee, isReceived)}
      >
        {!!title && (
          <DetailRow color={titleColor}>
            {title}
            {!!actionIcon && <ActionIcon name={actionIcon} iconColor={statusIconColor} />}
          </DetailRow>
        )}
        {customActionTitle}
        {!!subtitle && <DetailParagraph>{subtitle}</DetailParagraph>}
        {!!errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
      </DetailModal>
    );